File: src/main/java/com/growingio/kafka/TimestampComparator.java
package com.growingio.kafka;

/**
 * Created by foolchi on 01/08/15.
 */
public class TimestampComparator implements BinaryComparator {
    public int compare(String msg, String dest) {
        long diff = Long.parseLong(msg) - Long.parseLong(dest);
        if (diff > 0)
            return 1;
        if (diff < 0)
            return -1;
        return 0;
    }
}

File: src/main/scala/com/growingio/kafka/TimestampComparator.scala
package com.growingio.kafka

/**
 * Created by foolchi on 03/08/15.
 * 简单的时间戳比较类
 */
class TimestampComparator (dest : String) extends BinaryComparator{
  private val destTime = dest.toLong

  override def compare(msg: String): Int = {
    val diff = msg.toLong - destTime
    if (diff > 0) 1 else if (diff < 0) -1 else 0
  }
}
